go_backend/main.go

```diff
@@ -0,0 +1,242 @@
+package main
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/gin-gonic/gin"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func homePage(c *gin.Context) {
+	c.JSON(200, gin.H{
+		"message": "Hello World",
+	})
+}
+
+// func postHomePage(c *gin.Context) {
+// 	body := c.Request.Body
+// 	value, err := ioutil.ReadAll(body)
+// 	if err != nil {
+// 		fmt.Println(err.Error())
+// 	}
+// 	c.JSON(200, gin.H{
+// 		"message": string(value),
+// 	})
+// }
+
+//sign up function
+func querySignUp(c *gin.Context) {
+	//getting data from request
+	// userID := c.Query("userID")
+	email := c.Query("email")
+	password := c.Query("password")
+	firstName := c.Query("firstName")
+	lastName := c.Query("lastName")
+	phoneNumber := c.Query("phoneNumber")
+	userIdentity := c.Query("userIdentity")
+
+	//connect to the db
+	db, err := sql.Open("mysql", "root:@tcp(127.0.0.1:3306)/foodapp")
+	if err != nil {
+		fmt.Println("DB connection error")
+		c.JSON(200, gin.H{
+			"httpCode": "500",
+			"message":  "Database is not connected",
+		})
+		panic(err.Error())
+	} else {
+		fmt.Println("DB Connected!")
+	}
+	defer db.Close()
+
+	//insert to db
+	insert, err := db.Query("INSERT INTO userinfo VALUES(DEFAULT,?,?,?,?,?,?)", email, password, firstName, lastName, phoneNumber, userIdentity)
+
+	if err != nil {
+		fmt.Println("Sign up error")
```

Comment:
User might want to know why the signup is not vailed.

Reply:
normally, a single message "Your username or password is invalid" for secure purpose, I can change "Sign up error" to that if you think its ok